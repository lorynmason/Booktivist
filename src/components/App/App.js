import React, { Component } from 'react';
import { Header } from '../Header/Header'
import '../../styles/main.scss'

class App extends Component {

  componentDidMount() {
    //check local storage
  }

  render() {
    return (
      <div className="App">
        <Header />
      </div>
    );
  }
}

export default App;
