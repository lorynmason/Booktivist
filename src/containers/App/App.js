import React, { Component } from 'react';
import Header from '../../containers/Header/Header'
import CardContainer from '../CardContainer/CardContainer'
import '../../styles/main.scss'
import { connect } from 'react-redux';
import { Switch, Route, Redirect, withRouter } from 'react-router';

class App extends Component {

  componentDidMount() {
    //check local storage
  }

  render() {
    return (
      <div className="App">
        <Route path="/" component={Header} />
        <Route path="/SearchResults" component={CardContainer} />
        <Route path="/MustReadList" component={CardContainer} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
});

export const mapDispatchToProps = dispatch => ({
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(App));
