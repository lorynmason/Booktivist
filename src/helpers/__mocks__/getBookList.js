export const getBookList = jest.fn();