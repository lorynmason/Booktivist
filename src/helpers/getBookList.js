import { retrieveBookList } from '../actions'

export const getBookList = () => {
  return dispatch => {
    const bookList = JSON.parse(localStorage.getItem('bookList'))
    if (bookList) {
      dispatch(retrieveBookList(bookList))
    }
  }
}