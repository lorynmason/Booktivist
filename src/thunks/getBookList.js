import { retrieveBookList } from '../actions'

export const getBookList = () => {
  return dispatch => {
    const bookList = JSON.parse(localStorage.getItem('bookList'))
    dispatch(retrieveBookList(bookList))
  }
}