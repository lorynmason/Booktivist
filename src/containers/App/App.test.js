import React from 'react';
import { App, mapDispatchToProps} from './App';
import { shallow } from 'enzyme'
import { getBookList } from '../../helpers/getBookList'

describe('App', () => {
  let wrapper;
  const mockFunc = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<App getBookList={mockFunc}/>)
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call getBookList on componentDidMount', () => {
    expect(mockFunc).toHaveBeenCalled()
  })

  it.skip('mapDispatchToProps should dispatch getBookList helper when getBookList is called from props', () => {
    jest.mock('../../helpers/getBookList');
    const mockDispatch = jest.fn();
    getBookList.mockImplementation(() => {})

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getBookList();

    expect(mockDispatch).toHaveBeenCalled();
  })
})
