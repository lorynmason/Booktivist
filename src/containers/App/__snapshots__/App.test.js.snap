// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    getBookList={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
        />,
        <Switch>
          <Route
            component={[Function]}
            path="/SearchResults"
          />
          <Route
            component={[Function]}
            path="/MustReadList"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/SearchResults"
            />,
            <Route
              component={[Function]}
              path="/MustReadList"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/SearchResults",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/MustReadList",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
          />,
          <Switch>
            <Route
              component={[Function]}
              path="/SearchResults"
            />
            <Route
              component={[Function]}
              path="/MustReadList"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/SearchResults"
              />,
              <Route
                component={[Function]}
                path="/MustReadList"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/SearchResults",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/MustReadList",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
